File: SANGNT.py
n = int(input())

arr = []
is_prime = [True] * (n + 1)
is_prime[0] = is_prime[1] = False

for i in range(2, n + 1):
    if is_prime[i]:
        arr.append(i)
        for j in range(i*i, n+1, i):
            is_prime[j] = False

print(*arr)

File: SANGNT/SANGNT.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> arr;
    vector<bool> is_prime(n + 1, true);
    is_prime[0] = is_prime[1] = false;
    for (int i = 2; i <= n; i++) {
        if (is_prime[i]) {
            arr.push_back(i);
            for (int j = i * i; j <= n; j += i) {
                is_prime[j] = false;
            }
        }
    }
    for (int i = 0; i < arr.size(); i++) {
        cout << arr[i];
        if (i < arr.size() - 1) cout << " ";
    }
    cout << endl;
    return 0;
}
